pages/mesosphere/dcos/2.1/security/ent/tls-ssl/ar-custom/index.md

```diff
@@ -0,0 +1,243 @@
+
+- **Private key associated with the custom external certificate:** A private key in the PKCS#8 format associated with the custom external certificate.
+
+- **Installation directory:** The directory on the bootstrap node where the DC/OS installer resides. It is denoted with `$DCOS_INSTALL_DIR` in this document.
+
+- **Configuration:** The set of the configuration parameters that governs the specific aspects of the installation procedure. The configuration is stored in the DC/OS configuration file.
+
+- **DC/OS configuration file:** The file which contains the DC/OS configuration parameters. The DC/OS configuration file is normally called `config.yaml` and must be present in the `$DCOS_INSTALL_DIR/genconf/` directory on the bootstrap node during the installation. It is used by the DC/OS installer.
+
+
+# Requirements
+
+In order to install DC/OS Enterprise with a custom external certificate you will need:
+
+- to use the [advanced DC/OS installation method](/mesosphere/dcos/2.1/installing/ent/custom/advanced/). Other installation methods are not supported.
+- A file containing the custom external certificate.
+- A file containing the private key associated with the custom external certificate.
+
+## <a name="manually-placing-custom"></a>Manually placing custom external certificate
+
+
+The custom external certificate and the associated private key must be put in the `$DCOS_INSTALL_DIR/genconf/` directory on the bootstrap node:
+
+```bash
+cd $DCOS_INSTALL_DIR
+ls genconf/
+```
+```bash
+external-certificate.crt
+external-certificate-key.key
+```
+
+## <a name="manually-placing-master"></a>Manually placing the private key 
+
+
+For security reasons, the installer will not copy the private key from the bootstrap node to the master nodes.
+The private key associated with the custom external certificate must be securely distributed manually to every DC/OS master node **before starting the installation**.
+
+The filesystem path for the private key file must be `/var/lib/dcos/pki/tls/private/adminrouter-external.key`.
+The directory `/var/lib/dcos/pki/tls/private/` can be created manually with the following command before putting the file `adminrouter-external.key` in the directory on every DC/OS master node:
+
+```bash
+ mkdir -p /var/lib/dcos/pki/tls/private/
+```
+
+Furthermore, the file containing the private key `adminrouter-external.key` corresponding to the custom external certificate must be owned by the root Unix user and have 0600 permissions set.
+
+If you copy the private key file over the network onto the master nodes, the network channel must be adequately protected. An example of copying the external certificate private key is given below. The commands are executed on the bootstrap node. The `W.X.Y.Z` below indicates the IP address of a master node:
+
+```bash
+cd $DCOS_INSTALL_DIR/genconf
+scp external-certificate-key.key centos@W.X.Y.Z:/var/lib/dcos/pki/tls/private/adminrouter-external.key
```

Comment:
This command will fail since the `centos` user does not have permission to write to the directory.

Reply:
Fixed to avoid this problem